main.cpp

```diff
@@ -42,16 +42,64 @@ struct Point
     }
 private:
     float x{0}, y{0};
-};
+};
+
+
 
 template<typename Type>
 struct Wrapper
 {
     Wrapper(Type&& t) : val(std::move(t)) 
     { 
         std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl; 
-    }
-};
+    }
+
+    void print()
+    {
+        std::cout << "Wrapper::print(" << val << ")" << std::endl;
+    }
+
+    Type val{}; 
+    
+};
+// 8) You will need to specialize the Wrapper class template to work with the Point class.
+//a separate class definition entirely
+#include <memory>
+#
+template<>
+struct Wrapper<Point>
+{
+    using Type = Point;
+    Wrapper(Type&& v) : v_(std::move(v))
+    {
+        std::cout << "Wrapper(" << typeid(v_).name() << ")" << std::endl; 
+    }
+
+    void print()
+    {
+        std::cout << "Wrapper::print(" << v_.toString() << ")" << std::endl;
+    }
+
+    Type v_{0.f, 0.f};
+
+};
+
+
+template<typename T, typename ...Args>
+void variadicHelper(T first, Args ... everythingElse)
+{
+    Wrapper<T> (std::forward<T> (first)).print();
+    // Member ( std::forward<Args>(args)… ) {}
+    variadicHelper (std::forward<Args>(everythingElse) ...);
+    
+}
+
+
+template<typename T>
+void variadicHelper(T&& singleParam)
+{
+    Wrapper<T> (std::forward<T> (singleParam)).print();
+}

```

Comment:
Project9.1 challenge 2:
There is another recursive solution.  it is possible for `Args ...` to be empty.
your task: remove your implementation for instruction `4)` and implement this alternate solution.
implement it **after** your existing `variadicHelper` function.
hint: what does the **recursive call** actually look like when called with an empty Variadic Parameter list?